Vagrantfile

```diff
@@ -0,0 +1,60 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. 
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+  config.vm.network "forwarded_port", guest: 5000, host: 5000
+  config.vm.provision "shell", inline: <<-SHELL 
+    sudo echo "nameserver 8.8.8.8" >> /etc/resolv.conf
+    set -ex
+  SHELL
+
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL  
+    set -ex
+
+    # Install pyenv prerequisites
+    sudo apt-get update
+    sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+      libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+
+    # Install pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+    echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n eval "$(pyenv init --path)"\n fi' >> ~/.profile
+
+    source ~/.profile
+
+    /home/vagrant/.pyenv/bin/pyenv install 3.9.5
+    /home/vagrant/.pyenv/bin/pyenv global 3.9.5
+
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+    echo 'export PATH="$HOME/.poetry/bin:$PATH"' >> ~/.profile
+    echo 'export PATH="/vagrant:$PATH"' >> ~/.profile
+    source $HOME/.poetry/env
+
+  SHELL
+
+  $run_app = <<-SCRIPT
+    cd /vagrant 
+    echo $pwd
+    poetry install 
+    echo 'poetry installed'
+    pip install gunicorn
```

Comment:
Instead of using pip to install gunicorn, it will be simpler to use Poetry. Just add it via `poetry add gunicorn` on your computer so pyproject.toml is updated.

Then it will be installed on the VM during `poetry install`.

To run the app via gunicorn instead of "flask run", you first need to set environment variables based on the .env file (where as "flask run" does that for you)
```
export `cat .env | grep '^[A-Z]' | xargs`
```
Then replace "flask run --host=0.0.0.0" with your gunicorn command.

The gunicorn command takes the form of `gunicorn your_module:the_actual_app`.

In this case, "your_module" is `todo_app.app` and the_actual_app is `create_app()`. But brackets are a special character in bash so you will want to wrap it in quotes: `"todo_app.app:create_app()"`
The gunicorn equivalent to `--host=0.0.0.0` is `--bind 0.0.0.0:5000`